File: Ecosystem/Herbivore.m
classdef Herbivore < Creature
    %Creature which only eats plants
    properties
       
    end
    methods
        function obj = Herbivore(location)
            obj.location = location;
            obj.regen = 0.05;
            obj.damage = 0.1;
            obj.type = "herbivore";
            obj.enemy_type = "predator";
            obj.food_type = "plant";
        end
        
    end
end

File: Ecosystem_Python/herbivore.py
from creature import Creature


class Herbivore(Creature):
    """Creature which only eats plants"""
    
    def __init__(self, location):
        super().__init__(location)
        self.type = "herbivore"
        self.enemy_type = "predator"
        self.food_type = "plant"
        self.regen = 0.05
        self.damage = 0.1